src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegion.java

```diff
@@ -304,9 +307,22 @@ public void addAll(final Collection<GATKRead> readsToAdd){
         Utils.validateArg( padding >= 0, () -> "padding must be a positive integer but got " + padding);
         Utils.validateArg( genomeLoc.size() > 0, () -> "GenomeLoc must have size > 0 but got " + genomeLoc);
 
-        return referenceReader.getSubsequenceAt( genomeLoc.getContig(),
+        final String contig = genomeLoc.getContig();
+        final SAMSequenceRecord sequence = Utils.nonNull(referenceReader.getSequenceDictionary().getSequence(contig),
+                () -> "Contig " + contig + " not found in reference. " +
+                        "\nPlease check that you are using a compatible reference for your data." +
+                        "\nReference Sequences: " + listContigsAsString(referenceReader.getSequenceDictionary()));
+
+        return referenceReader.getSubsequenceAt(contig,
                 Math.max(1, genomeLoc.getStart() - padding),
-                Math.min(referenceReader.getSequenceDictionary().getSequence(genomeLoc.getContig()).getSequenceLength(), genomeLoc.getEnd() + padding) ).getBases();
+                Math.min(sequence.getSequenceLength(), genomeLoc.getEnd() + padding)).getBases();
+    }
+
+    private static String listContigsAsString(final SAMSequenceDictionary sequenceDictionary) {
```

Comment:
Move to `SequenceDictionaryUtils` (or, better, use one of the existing methods there like `SequenceDictionaryUtils.getContigNamesList()`)

Reply:
Of course that exists already.

Reply:
It actually exists in at least 2 places.  ReadUtils.prettyPrintSequenceRecords seems to be the version I want.